src/bridge/setup.rs

```diff
@@ -18,30 +50,12 @@ pub async fn setup_neovide_remote_clipboard(nvim: &Neovim<TxWrapper>, neovide_ch
         return;
     }
 
-    // don't know how to setup lambdas with Value, so use string as command
-    let custom_clipboard = r#"
-        let g:clipboard = {
-          'name': 'custom',
-          'copy': {
-            '+': {
-              lines,
-              regtype -> rpcnotify(neovide_channel, 'neovide.set_clipboard', lines, regtype, '+')
-            },
-            '*': {
-              lines,
-              regtype -> rpcnotify(neovide_channel, 'neovide.set_clipboard', lines, regtype, '*')
-            },
-          },
-          'paste': {
-            '+': {-> rpcrequest(neovide_channel, 'neovide.get_clipboard', '+')},
-            '*': {-> rpcrequest(neovide_channel, 'neovide.get_clipboard', '*')},
-          },
-          'cache_enabled': 0
-        }
-        "#
-    .replace('\n', "") // make one-liner, because multiline is not accepted (?)
-    .replace("neovide_channel", &neovide_channel.to_string());
-    nvim.command(&custom_clipboard).await.ok();
+    nvim.set_var("neovide_channel_id", Value::from(neovide_channel))
```

Comment:
If you set this var, shouldn't it be removed after executing the Lua code? Else someone could stumble across this and potentially rely on an outdated value.

Reply:
Unfortunately, that won't work, since the variable is read whenever the `+` and `*` registers are accessed. Reading the channel from a variable allows us to use the same clipboard provider across multiple UI sessions.

> Else someone could stumble across this and potentially rely on an outdated value.

A session has only one channel id in its lifetime, and the variable is updated whenever Neovide attaches to nvim, so this shouldn't be too much of an issue. (Unless the user starts multiple overlapping sessions, but we don't seem to support that at all). 

While I'm generally strongly in favor of data encapsulation, having the channel id in a global variable makes debugging of channel issues a lot easier.

An alternative approach is to forcibly reload the clipboard provider whenever Neovide attaches to nvim. We might need something like that in order to fix #1331. This could also make it possible to mix different UI clients without breaking the clipboard. The drawbacks are that this seems like a hack, and it might break things for users who rely on the clipboard provider being loaded only once. I think the best approach is to make forcible reloading optional (can be disabled by setting a variable) and combining it with the present solution.

Reply:
That's a good idea, thanks for the extended explanation!